tasks/build_tags.py

```diff
@@ -53,3 +53,43 @@ def get_build_tags(include, exclude):
     include = ALL_TAGS.intersection(set(include))
     exclude = ALL_TAGS.intersection(set(exclude))
     return list(include - exclude)
+
+
+@task
+def audit_tag_impact(ctx, build_exclude=None, use_embedded_libs=False):
+    """
+    Measure each tag's contribution to the binary size
+    """
+    build_exclude = [] if build_exclude is None else build_exclude.split(",")
+
+    tags_to_audit = ALL_TAGS.difference(set(build_exclude)).difference(set(PUPPY_TAGS))
+
+    max_size = _compute_build_size(ctx, build_exclude=','.join(build_exclude), use_embedded_libs=use_embedded_libs)
+    print("size with all tags is {} kB".format(max_size / 1000))
+
+    puppy_size = _compute_build_size(ctx, puppy=True, use_embedded_libs=use_embedded_libs)
+    print("puppy size is {} kB\n".format(puppy_size / 1000))
+
+    report = {"unaccounted": max_size - puppy_size, "puppy": puppy_size}
+
+    for tag in tags_to_audit:
+        exclude_string = ','.join(build_exclude + [tag])
+        size = _compute_build_size(ctx, build_exclude=exclude_string, use_embedded_libs=use_embedded_libs)
+        delta = (max_size - size)
+        print("tag {} adds {} kB (excludes: {})".format(tag, delta / 1000, exclude_string))
+        report[tag] = delta
+        report["unaccounted"] -= delta
+
+    print("\nCSV output in bytes:")
```

Comment:
I think the CSV output could be behind a flag, and you either one or the other

Reply:
As the total run time on my machine is 9 minutes, having a progress indicator seems necessary. but indeed I added the `--csv` flag add, disabled by default.